pkgs/development/libraries/onnxruntime/default.nix

```diff
@@ -30,11 +30,17 @@ let
     sha256 = "sha256-BYL7wxsYRI45l8C3VwxYIIocn5TzJnBtU0UZ9pHwwZw=";
   };
 
-  eigen = fetchFromGitLab {
-    owner = "libeigen";
-    repo = "eigen";
-    rev = "d10b27fe37736d2944630ecd7557cefa95cf87c9";
-    sha256 = "sha256-Lmco0s9gIm9sIw7lCr5Iewye3RmrHEE4HLfyzRkQCm0=";
+  eigen' = applyPatches {
+    name = "patch-build-error.patch";
+    src = eigen.src;
+    patches = eigen.patches or [] ++ [
+      # Patch for Eigen seems to be necessary to avoid build errors as referenced here:
+      # https://aur.archlinux.org/pkgbase/python-onnxruntime?O=30#comment-856116
+      (fetchpatch {
+        url = "https://gitlab.com/libeigen/eigen/-/commit/d0e3791b1a0e2db9edd5f1d1befdb2ac5a40efe0.patch";
+        sha256 = "sha256-8qiNpuYehnoiGiqy0c3Mcb45pwrmc6W4rzCxoLDSvj0=";
+      })
+    ];
```

Comment:
This seems fragile; it'd break whenever eigen is updated.

Reply:
@Atemu What is the alternative? Do you want `src` to be defined inline and pinned to a specific `rev`? I see this existing thing done in multiple places in Nixpkgs, we'd be warned if the hunks fail to apply, then we know to go fix the patch.

Reply:
Why isn't that patch applied to the eigen package?

Reply:
> Date: Wed, 20 Oct 2021 16:03:12 -0700
> Subject: [PATCH] Fix vectorized reductions for Eigen::half

Also how come this hasn't been merged upstream?

Reply:
It has: https://gitlab.com/libeigen/eigen/-/merge_requests/699

Reply:
After 2 years there hasn't been a release of eigen it seems :disappointed:

The eigen package itself builds fine. It doesn't make sense to apply a patch there if it has no build problems. :shrug:

Reply:
> Why isn't that patch applied to the eigen package?

Because onnxruntime consumes the patched source, which we can't get from a patched eigen derivation.

Reply:
> Because onnxruntime consumes the patched source, which we can't get from a patched eigen derivation.

It needn't to. It should use FIND_PACKAGE_ARGS instead

Reply:
https://github.com/search?q=repo%3Amicrosoft%2Fonnxruntime%20FIND_PACKAGE_ARGS&type=code

They're using that for *some* dependencies.

Reply:
> After 2 years there hasn't been a release of eigen it seems 😞

Oh boi https://gitlab.com/libeigen/eigen/-/issues/2699